PLT-6763 FE: Auditors sign certificates

amnambiar opened the pull request:
https://input-output.atlassian.net/browse/PLT-6763


Across BE task - [PLT-6367](https://input-output.atlassian.net/browse/PLT-6367) / [PR #74](https://github.com/input-output-hk/dapps-certification/pull/74) (merged into `master`)

BE Revised fixes - feat/l1-certificate [PR #90](https://github.com/input-output-hk/dapps-certification/pull/90)

@coderabbitai:ignore

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
### Summary by CodeRabbit

- New Feature: Introduced `CertificationForm` component for dynamic certification form handling.
- New Feature: Added `CertificationMetadata` component to submit and manage certification metadata.
- New Feature: Integrated `CertificationMetadata` into `CreateCertificate` component, allowing users to review and submit certification metadata.
- Refactor: Updated `ReportUpload` component to use the new `CertificationForm` component, improving code modularity and maintainability.
- Refactor: Enhanced `CreateCertificate` component with improved state management and error handling.
- Style: Adjusted CSS styles across various components for better UI consistency.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

@coderabbitai: ignore

amnambiar committed:
WIP

amnambiar committed:
PLT-6763 Auditor L1 Certificate metadata form - optimized auditor report upload form

amnambiar committed:
PLT-6763 Fixed API

amnambiar committed:
PLT-6763 Minor fixes

amnambiar commented on src/pages/certification/components/certificationMetadata/CertificationMetadata.tsx:
```diff
@@ -0,0 +1,154 @@
+        name: name,
+        logo: logo || "",
+        social: {
+          contact: email,
+          discord: discord || "",
+          twitter: twitter || "",
+          github: github || "",
+          website: website,
+        },
+      },
+      disclaimer: disclaimer,
+      scripts: formattedDappScripts,
+      subject: subject,
+      summary: summary,
+    };
+
+    setSubmitting(true);
+
+    const response: any = await fetchData
+      .post("/run/" + uuid + "/certificate", payload)
+      .catch((errorObj) => {
+        let errorMsg = "Something went wrong. Please try again.";
+        if (errorObj?.response?.data) {
+          errorMsg =
+            errorObj.response.statusText + " - " + errorObj.response.data;
+        }
+        setShowError(errorMsg);
+        const timeout = setTimeout(() => {
+          clearTimeout(timeout);
+          setShowError("");
+        }, 5000);
+        setSubmitting(false);
+      });
+
+    setShowError("");
+    setOpenModal(true);
+    exportObjectToJsonFile(response.data.offchain, "Off-Chain_" + uuid + ".json");
+    exportObjectToJsonFile(response.data.onchain, "On-Chain_" + uuid + ".json");
+    setSubmitting(false);
+  };
```
updated the block in a .then

amnambiar replied:
undid the change and Ignoring the suggestion for now.

amnambiar commented on src/pages/certification/components/certificationMetadata/CertificationMetadata.tsx:
```diff
@@ -0,0 +1,154 @@
+        />
+      </div>
+
+      {showError ? <Toast message={showError} /> : null}
+
+      <Modal
```
ignoring for now

amnambiar commented on src/components/CreateCertificate/CreateCertificate.tsx:
```diff
@@ -128,23 +153,43 @@ const CreateCertificate = () => {
-        {showError ? <Toast message={showError} /> : null}
-    </>);
+    return (
+        <>
+            {subscribedFeatures?.indexOf("l2-upload-report") === -1 ? <>
+                {certified || disableCertify ? null : (<Button
+                    displayStyle="gradient"
+                    onClick={() => triggerGetCertificate()}
+                    buttonLabel={"Purchase a Certificate" + (certificationPrice ? " (" + (certificationPrice / 1000000).toString() + " ADA)" : "")}
+                    showLoader={certifying}
+                />)}
+                {transactionId ? (
+                    <Modal open={openModal} title="Certification Successful" onCloseModal={onCloseModal}>
+                        <span>
+                            View your certification broadcasted on-chain&nbsp;
+                            <a target="_blank" rel="noreferrer" href={`https://preprod.cardanoscan.io/transaction/${transactionId}`}>here</a>!
+                        </span>
+                    </Modal>
+                ) : null}
+            </> : <>
+                <Button
+                    displayStyle="gradient"
+                    onClick={() => setMetadataModalStatus(true)}
+                    buttonLabel={"Review Certification Metadata"}
+                    showLoader={certifying}
+                />
+                <Modal
+                    open={openMetadataModal}
+                    title="Certification Metadata"
+                    className="certification-metadata"
+                    onCloseModal={()=> {}}
+                    fullWidth
+                >
+                    <CertificationMetadata uuid={uuid} onCloseForm={onCloseMetadataForm}/>
+                </Modal>
+            </>}
+
+            {showError ? <Toast message={showError} /> : null}
+        </>);
 }
```
ignoring for now

amnambiar commented on src/components/CertificationForm/CertificationForm.tsx:
```diff
@@ -0,0 +1,203 @@
+  addScripts: { scriptFields: IInfo[]; smartContractInfo: IInfo[] };
+}
+
+const Component = (props: FieldConfig) => {
+  const { id, ...rest } = props.config;
+
+  switch (props.componentType) {
+    case "text":
+      return (
+        <Input
+          id={props.config.id}
+          {...props.form.register(props.config.id)}
+          {...rest}
+        />
+      );
+    case "textarea":
+      return <TextArea {...props.form.register(props.config.id)} {...rest} />;
+    case "dropdown":
+      return (
+        <Dropdown
+          onOptionSelect={(option) =>
+            props.form.setValue(props.config.id, option.value, {
+              shouldValidate: true, // trigger on change validation manually
+            })
+          }
+          {...props.form.register(props.config.id)}
+          {...rest}
+        />
+      );
+    default:
+      return null;
+  }
+};
```
ignoring for now

amnambiar commented on src/components/CertificationForm/CertificationForm.tsx:
```diff
@@ -0,0 +1,203 @@
+    );
+  };
+
+  const initializeFormState = () => {
+    form.clearErrors(); // clear form errors
+    form.reset(initData);
+  };
+
+  useEffect(() => {
+    initializeFormState();
+    // initializeFormState() is to not to be triggered on every re-render of the dep-array below but whenever the form or userDetails is updated alone
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [form]);
```
ignoring this for now; will recheck suggestion after revamp

amnambiar committed:
PLT-6763 AI review comments

amnambiar committed:
PLT-6763 AI Review comment

amnambiar commented on src/components/CreateCertificate/CreateCertificate.tsx:
```diff
@@ -67,7 +96,7 @@ const CreateCertificate = () => {
         } else if (errorObj?.response?.message) {
             errorMsg = errorObj?.response.message + ' Please try again.'
         } else if (errorObj?.response?.data) {
-            errorMsg = errorObj.response.statusText + ' - ' + errorObj.response.data 
+            errorMsg = errorObj.response.statusText + ' - ' + errorObj.response.data
         }
         setShowError(errorMsg.length > 50 ? 'Something wrong occurred. Please try again later.' : errorMsg);
         const timeout = setTimeout(() => { clearTimeout(timeout); setShowError("") }, 5000)
```
ignoring

amnambiar commented on src/components/CreateCertificate/CreateCertificate.tsx:
```diff
@@ -5,59 +5,88 @@ import { useAppSelector } from "store/store";
-    const [ showError, setShowError ] = useState("");
-    const [ openModal, setOpenModal ] = useState(false);
-    const [ disableCertify, setDisableCertify ] = useState(false);
+    const { address, wallet, subscribedFeatures } = useAppSelector(
+        (state) => state.auth
+    );
```
ignoring

amnambiar commented on src/components/CertificationForm/CertificationForm.tsx:
```diff
@@ -0,0 +1,203 @@
+    form.reset(initData);
+  };
+
+  useEffect(() => {
+    initializeFormState();
+    // initializeFormState() is to not to be triggered on every re-render of the dep-array below but whenever the form or userDetails is updated alone
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [form]);
```
ignoring

amnambiar commented on src/components/CertificationForm/CertificationForm.tsx:
```diff
@@ -0,0 +1,203 @@
+  addScripts: { scriptFields: IInfo[]; smartContractInfo: IInfo[] };
+}
+
+const Component = (props: FieldConfig) => {
+  const { id, ...rest } = props.config;
+
+  switch (props.componentType) {
+    case "text":
+      return (
+        <Input
+          id={props.config.id}
+          {...props.form.register(props.config.id)}
+          {...rest}
+        />
+      );
+    case "textarea":
+      return <TextArea {...props.form.register(props.config.id)} {...rest} />;
+    case "dropdown":
+      return (
+        <Dropdown
+          onOptionSelect={(option) =>
+            props.form.setValue(props.config.id, option.value, {
+              shouldValidate: true, // trigger on change validation manually
+            })
+          }
+          {...props.form.register(props.config.id)}
+          {...rest}
+        />
+      );
+    default:
+      return null;
+  }
+};
```
ignoring

amnambiar commented on src/pages/certification/components/certificationMetadata/CertificationMetadata.tsx:
```diff
@@ -0,0 +1,153 @@
+        />
+      </div>
+
+      {showError ? <Toast message={showError} /> : null}
```
ignoring

amnambiar commented on src/pages/certification/components/certificationMetadata/CertificationMetadata.tsx:
```diff
@@ -0,0 +1,153 @@
+        name: name,
+        logo: logo || "",
+        social: {
+          contact: email,
+          discord: discord || "",
+          twitter: twitter || "",
+          github: github || "",
+          website: website,
+        },
+      },
+      disclaimer: disclaimer,
+      scripts: formattedDappScripts,
+      subject: subject,
+      summary: summary,
+    };
+
+    setSubmitting(true);
+
+    const response: any = await fetchData
+      .post("/run/" + uuid + "/certificate", payload)
+      .catch((errorObj) => {
+        let errorMsg = "Something went wrong. Please try again.";
+        if (errorObj?.response?.data) {
+          errorMsg =
+            errorObj.response.statusText + " - " + errorObj.response.data;
+        }
+        setShowError(errorMsg);
+        const timeout = setTimeout(() => {
+          clearTimeout(timeout);
+          setShowError("");
+        }, 5000);
+      });
+      if (!showError) {
+        setShowError("");
+        setOpenModal(true);
+        exportObjectToJsonFile(response.data.offchain, "Off-Chain_" + uuid + ".json");
+        exportObjectToJsonFile(response.data.onchain, "On-Chain_" + uuid + ".json");
+      }
+    setSubmitting(false);
+  };
```
ignoring for now

amnambiar commented on src/pages/certification/components/certificationMetadata/CertificationMetadata.tsx:
```diff
@@ -0,0 +1,153 @@
+    onCloseForm()
+  };
+
+  const transformDappScripts = (scripts: []) => {
+    return scripts.map((script: any) => {
+      const { scriptHash, contractAddress, ...rest } = script;
+      return({
+        scriptHash: scriptHash,
+        contractAddress: contractAddress,
+        smartContractInfo: {
+          ...rest,
+        },
+      });
+    });
+  };
```
ignoring

amnambiar committed:
PLT-6763 AI review comments

amnambiar commented on src/pages/certification/components/certificationMetadata/CertificationMetadata.tsx:
```diff
@@ -0,0 +1,153 @@
+        setOpenModal(true);
+        exportObjectToJsonFile(response.data.offchain, "Off-Chain_" + uuid + ".json");
+        exportObjectToJsonFile(response.data.onchain, "On-Chain_" + uuid + ".json");
+      }
```
ignoring the suggestion for now.

amnambiar commented on src/components/CreateCertificate/CreateCertificate.tsx:
```diff
@@ -5,59 +5,89 @@ import { useAppSelector } from "store/store";
+        Promise.all([
+            fetchData.get("/profile/current/balance"),
+            fetchData.get("/run/" + uuid + "/details")
+        ])
+            .then(([balanceResponse, runDetailsResponse]) => {
+                const availableProfileBalance: number = balanceResponse.data;
+                const runDetails: Run = runDetailsResponse.data;
                 setCertificationPrice(runDetails.certificationPrice);
-                setPerformTransaction((availableProfileBalance >= 0 && (availableProfileBalance - runDetails.certificationPrice) < 0) ? true : false)
+                setPerformTransaction(
+                    availableProfileBalance >= 0 &&
+                        availableProfileBalance - runDetails.certificationPrice < 0
+                        ? true
+                        : false
+                );
             })
-        })
+            .catch((error) => {
+                console.error(error);
+            });
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [])
-    
+    }, []);
```
ignoring

amnambiar commented on src/components/CreateCertificate/CreateCertificate.tsx:
```diff
@@ -128,23 +158,43 @@ const CreateCertificate = () => {
+            </>}
+
+            {showError ? <Toast message={showError} /> : null}
+        </>);
```
ignoring as already done

bogdan-manole commented:
After I bought the subscription, I had to refresh  several times to see the Auditing menu

bogdan-manole commented:
Validation error in the case of valid value
<img width="982" alt="image" src="https://github.com/input-output-hk/dapps-certification-web/assets/3176255/97792c48-e7d6-4d8b-b421-8a0cbd90063a">

bogdan-manole commented:
I wasn't able to submit it even if the form didn't show any validation error or unfilled mandatory field


https://github.com/input-output-hk/dapps-certification-web/assets/3176255/f238c1a8-1a50-4086-9383-f071739a8855

bogdan-manole commented:
Subject inpute accepts in the field invalid format

<img width="921" alt="image" src="https://github.com/input-output-hk/dapps-certification-web/assets/3176255/bf3195e7-7a09-4c4d-9831-f468e20acca2">

Please consult the `Swagger` api data model specification

<img width="805" alt="image" src="https://github.com/input-output-hk/dapps-certification-web/assets/3176255/87bc9878-7582-4707-af2a-4e332267c76d">

amnambiar commented on src/pages/certification/components/certificationMetadata/CertificationMetadata.tsx:
```diff
@@ -0,0 +1,152 @@
+      certificateIssuer: {
+        name: name,
+        logo: logo || "",
+        social: {
+          contact: email,
+          discord: discord || "",
+          twitter: twitter || "",
+          github: github || "",
+          website: website,
+        },
+      },
+      disclaimer: disclaimer,
+      scripts: formattedDappScripts,
+      subject: subject,
+      summary: summary,
+    };
+
+    setSubmitting(true);
+
+    const response: any = await fetchData
+      .post("/run/" + uuid + "/certificate", payload)
+      .catch((errorObj) => {
+        let errorMsg = "Something went wrong. Please try again.";
+        if (errorObj?.response?.data) {
+          errorMsg =
+            errorObj.response.statusText + " - " + errorObj.response.data;
+        }
+        setShowError(errorMsg);
+        const timeout = setTimeout(() => {
+          clearTimeout(timeout);
+          setShowError("");
+        }, 5000);
+        setSubmitting(false);
+      });
+    setShowError("");
+    setOpenModal(true);
+    exportObjectToJsonFile(response.data.offchain, "Off-Chain_" + uuid + ".json");
+    exportObjectToJsonFile(response.data.onchain, "On-Chain_" + uuid + ".json");
+    setSubmitting(false);
+  };
```
ignoring for now.

amnambiar commented on src/components/CreateCertificate/CreateCertificate.tsx:
```diff
@@ -5,59 +5,89 @@ import { useAppSelector } from "store/store";
+        Promise.all([
+            fetchData.get("/profile/current/balance"),
+            fetchData.get("/run/" + uuid + "/details")
+        ])
+            .then(([balanceResponse, runDetailsResponse]) => {
+                const availableProfileBalance: number = balanceResponse.data;
+                const runDetails: Run = runDetailsResponse.data;
                 setCertificationPrice(runDetails.certificationPrice);
-                setPerformTransaction((availableProfileBalance >= 0 && (availableProfileBalance - runDetails.certificationPrice) < 0) ? true : false)
+                setPerformTransaction(
+                    availableProfileBalance >= 0 &&
+                        availableProfileBalance - runDetails.certificationPrice < 0
+                        ? true
+                        : false
+                );
             })
-        })
+            .catch((error) => {
+                console.error(error);
+            });
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [])
-    
+    }, []);
```
ignoring this for now

amnambiar commented on src/components/CreateCertificate/CreateCertificate.tsx:
```diff
@@ -67,7 +97,7 @@ const CreateCertificate = () => {
         } else if (errorObj?.response?.message) {
             errorMsg = errorObj?.response.message + ' Please try again.'
         } else if (errorObj?.response?.data) {
-            errorMsg = errorObj.response.statusText + ' - ' + errorObj.response.data 
+            errorMsg = errorObj.response.statusText + ' - ' + errorObj.response.data
         }
         setShowError(errorMsg.length > 50 ? 'Something wrong occurred. Please try again later.' : errorMsg);
         const timeout = setTimeout(() => { clearTimeout(timeout); setShowError("") }, 5000)
```
ignoring this for now

amnambiar commented:
@bogdan-manole 

> After I bought the subscription, I had to refresh several times to see the Auditing menu

Already resolved with a temporary hack (since in revamp the handling of subscriptions will be moved to before entering the pages) in [PR#10 PLT-7439 - Fix for Blank page in the tool at times](https://github.com/input-output-hk/dapps-certification-web/pull/10/files#diff-a100b1085452cd95c277db284381de09095b6bdccecdce73773661f5b4c55f22R206-R221)

> Validation error in the case of valid value

:) See it isn't a valid value. A valid http(s):// URL must have .json or .pdf. 

> I wasn't able to submit it even if the form didn't show any validation error or unfilled mandatory field

Thought is was resolved, checking it, will resolve. _Issue - Script Hash not showing validation errors_

> Subject inpute accepts in the field invalid format

Per [PLT-4386](https://input-output.atlassian.net/browse/PLT-4386) - _Subject [1-64 chars]_, and also discussion with @RSoulatIOHK (missed to pass the info to you :pray: )
![image](https://github.com/input-output-hk/dapps-certification-web/assets/11058223/085f884f-0a99-4916-92e5-bd3355c19c23)

amnambiar commented on src/pages/certification/components/certificationMetadata/CertificationMetadata.tsx:
```diff
@@ -0,0 +1,152 @@
+
+    const response: any = await fetchData
+      .post("/run/" + uuid + "/certificate", payload)
+      .catch((errorObj) => {
+        let errorMsg = "Something went wrong. Please try again.";
+        if (errorObj?.response?.data) {
+          errorMsg =
+            errorObj.response.statusText + " - " + errorObj.response.data;
+        }
+        setShowError(errorMsg);
+        const timeout = setTimeout(() => {
+          clearTimeout(timeout);
+          setShowError("");
+        }, 5000);
+        setSubmitting(false);
+      });
```
ignoring for now

amnambiar committed:
PLT-6763 Fixed error display in Dapp Script fields; fixed error handling

amnambiar committed:
PLT-6763 Removed dependecy array

bogdan-manole commented:
> :) See it isn't a valid value. A valid http(s):// URL must have .json or .pdf.

an URL could be a PDF without being specified as so. Please follow the Regex pattern specified by the Swagger for that field or let's agree on changing in the BE, anyway both FE and BE should follow the same validation pattern

RSoulatIOHK commented on src/pages/auditing/reportUpload/reportUpload.schema.tsx:
```diff
@@ -56,24 +56,14 @@ export const reportUploadSchema = yup.object().shape({
       /^(https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|www\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9]+\.[^\s]{2,}|www\.[a-zA-Z0-9]+\.[^\s]{2,})$/,
```
Can we add that http and https is not mandatory? I feel like people will want to use "example.com" instead of having to put "https://example.com"

amnambiar replied:
done

RSoulatIOHK commented:
## Level 0 certification
I don't get L0 audit when submitting and auditor report for L0 audit
`{"1304":{"metadata":[["ipfs://bafkreiaerryd4ou6edxrygimjxzgtv4wyjmlhcqqtg47cosz5oygzdnf","le"]],"rootHash":"64346f7e71cdc0b790c3992114f3ab3b15dddb7e4b5ffc10269a06fbb5484828","schemaVersion":1,"subject":"test","type":{"action":"CERTIFY","certificateIssuer":"test","certificationLevel":"l2"}}}`

Instead I should get:
`{"1304":{"metadata":[["ipfs://bafkreiaerryd4ou6edxrygimjxzgtv4wyjmlhcqqtg47cosz5oygzdnf","le"]],"rootHash":"64346f7e71cdc0b790c3992114f3ab3b15dddb7e4b5ffc10269a06fbb5484828","schemaVersion":1,"subject":"test","type":{"action":"**AUDIT**","certificateIssuer":"test","certificationLevel":**0**}}}`

Note: `certificationLevel` should be an `int` with just the certification level and not a `str`.

## Level 1 certification
I get "Undefined" in on/off chain metadata 
![image](https://github.com/input-output-hk/dapps-certification-web/assets/117812549/54ca475a-c814-4299-b43c-a0c85e1109ce)

## Level 2 certification 
Everything looks OK. Just the need for BE to change `certificationLevel` which should be an `int` with just the certification level and not a `str`.

RSoulatIOHK commented on src/pages/auditing/reportUpload/reportUpload.schema.tsx:
```diff
@@ -56,24 +56,14 @@ export const reportUploadSchema = yup.object().shape({
       /^(https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|www\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9]+\.[^\s]{2,}|www\.[a-zA-Z0-9]+\.[^\s]{2,})$/,
       "Please verify the characters entered"
```
The error message is not really useful, maybe something like:
```suggestion
      "Please enter a valid URL."
```

amnambiar replied:
Done

RSoulatIOHK commented on src/pages/auditing/reportUpload/reportUpload.schema.tsx:
```diff
@@ -56,24 +56,14 @@ export const reportUploadSchema = yup.object().shape({
         .string()
         .required("This field is required")
+        .max(64, "Please enter upto 64 characters")
         .matches(/[0-9a-fA-F]{64}/, "Please verify the 64 characters entered"),
```
```suggestion
        .matches(/[0-9a-fA-F]{64}/, "Please enter a valid script hash."),
```

RSoulatIOHK replied:
Same for other error messages, maybe change to something more useful.

amnambiar replied:
Done

RSoulatIOHK commented:
In DAPP Script (which should be DApp script I think from IOG standard), there is no display when a regex fails.
![image](https://github.com/input-output-hk/dapps-certification-web/assets/117812549/5c4987ee-43c6-41a2-a1b0-959992712ee9)

But you can't submit the report so the check is actually done. I found that repository URL needs to be a valid URL and Script Hash needs to be a valid Hash.

Contract address seems to have no check.

Don't force the "@" for twitter handle. I know I'm the one that added it a long time ago. Don't blame me plz :'(

To Sum up
- [x] Change DAPP to DApp
- [ ] Display when a regex fails for DApp Scripts (Repository URL and Script hash look missing)
- [x] Check that Contract address is a valid address
- [x] Remove the needs for an "@" in the twitter handle.

RSoulatIOHK reviewed (changes requested):
Just a few more light changes.

amnambiar committed:
PLT-6763 Review fixes + validation pattern updates

amnambiar commented:
@bogdan-manole 

> https://github.com/input-output-hk/dapps-certification-web/pull/8#issuecomment-1713760452

Can you fix the validation in BE. The regex used in FE is `/^(https?:\/\/)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,255}\.[a-z]{2,6}(\b([-a-zA-Z0-9@:%_\+.~#()?&\/\/=]*))?\.(?:jpg|jpeg|png|gif|bmp|svg|webp|tiff|tif)$/`

> Website URL regex pattern

/^(https?:\/\/)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,255}\.[a-z]{2,6}(\b([-a-zA-Z0-9@:%_\+.~#()?&\/\/=]*))?$/

@RSoulatIOHK ,

> Check that Contract address is a valid address

_Contact Address_ under _DApp Script_ will be validated across pattern `^(addr_test1|addr1)[a-zA-Z0-9]{53,}$`; and will be **non-mandatory**

> Display when a regex fails for DApp Scripts (Repository URL and Script hash look missing)

Able to see errors now with a minor fix, unable to reproduce the scenario when the errors weren't visible. Can you recheck

https://github.com/input-output-hk/dapps-certification-web/assets/11058223/ee106ddd-2b8f-46f3-989a-c399163e516b


Note -
 - Received undefined in off-chain and on-chain downloaded JSONs - because the APIs returned 204 No Content and FE didn't handle it properly. I'm on it, will verify once the API service is up and running, and comment 
 - BE is pointed towards https://dapps-certification-web.scdev.aws.iohkdev.io/ now, and not excuse.ro ( it has latest profile/subscription API updates for the moment)

amnambiar committed:
PLT-6763 Fix to handle 204 in POST of l1-Certificate and l2-certificate + minor schema fix + renamed Reset to Cancel

amnambiar commented:
@RSoulatIOHK , 

> Received undefined in off-chain and on-chain downloaded JSONs - 
> because the APIs returned 204 No Content and FE didn't handle it properly. I'm on it, will verify once the API service is up and running, and comment

Resolved this issue - at least by handling the response, and throwing an error to retry (instead of downloading invalid JSONs)

PR is ready for re-review

RSoulatIOHK commented:
I'll pull it this morning for test. Which BE version should I be using?

RSoulatIOHK commented on src/pages/auditing/reportUpload/reportUpload.schema.tsx:
```diff
@@ -86,7 +80,7 @@ export const reportUploadSchema = yup.object().shape({
         .matches(
           /^(?:https?:\/\/)?(?:www\.)?github\.com\/[\w-]+\/[\w.-]+$/, {
-            message: "Please verify the characters entered",
+            message: "Please enter a valid github repo URL",
```
```suggestion
            message: "Please enter a valid GitHub repository URL",
```

RSoulatIOHK commented:
![image](https://github.com/input-output-hk/dapps-certification-web/assets/117812549/e0f01dd1-52bb-4104-af47-79f9a721ff24)

On a failed testing, we should not be able to review Certification metadata because we should not issue any.

However we should be able to view the full html report, we currently only view the failed unit tests:
![image](https://github.com/input-output-hk/dapps-certification-web/assets/117812549/fb0b3e66-1814-43ad-abec-7fa7c2df196d)
vs
![image](https://github.com/input-output-hk/dapps-certification-web/assets/117812549/69ea0b4a-6afd-4041-aeff-8ca8d8c1bc00)

@amnambiar, @EehMauro Do you want me to move those to Issues so we can revisit that after the revamp?

RSoulatIOHK commented on src/pages/auditing/reportUpload/reportUpload.schema.tsx:
```diff
@@ -15,34 +15,34 @@ export const reportUploadSchema = yup.object().shape({
-    .matches(/^[0-9a-zA-Z_]+$/, "Enter a valid subject name (a-z, A-Z, 0-9 or _ characters only)."),
+    .max(64, "Please enter a valid subject with upto 64 characters")
+    .matches(/^[0-9a-zA-Z_]+$/, "Please enter a valid subject (that has a-z, A-Z, 0-9 or _ characters only)."),
   name: yup.string().required("This field is required"),
```
Name should not be longer than 64 chars because it will go on-chain

RSoulatIOHK replied:
I've raised the issue to Bogdan for BE

bogdan-manole replied:
done

RSoulatIOHK commented:
I don't know if the _online deployment_ is the right BE but I can't get "L1" metadata. I get "Something went wrong message"

Also, when you get this message, you can never retry as it says " Forbidden - Certification already started". This should fix.

I can also get this as an issue to revisit after revamp

amnambiar committed:
Update src/pages/auditing/reportUpload/reportUpload.schema.tsx

Co-authored-by: RSoulatIOHK <117812549+RSoulatIOHK@users.noreply.github.com>

amnambiar committed:
PLT-6763 Limit name to be upto 64 characters

amnambiar committed:
Merge branch 'PLT-6763' of https://github.com/input-output-hk/dapps-certification-web into PLT-6763

amnambiar commented:
@RSoulatIOHK ,

>  "Something went wrong message"

Due to 204 no content in POST api. Must be because the BE API is not the right one. Neither of the deployments excuse.ro nor aws has the right branch for this. Bogdan needs to confirm the branch that this needs to be verified across.

> On a failed testing, we should not be able to review Certification metadata because we should not issue any.

Is it OK to ensure this in revamp?


> After something went wrong message,  you can never retry as it says " Forbidden - Certification already started". This should fix.

Need some clarifications over the workflow, will address in revamp once that is clear

> However we should be able to view the full html report, we currently only view the failed unit tests:

Will handle in revamp, as this will be a tabular view

amnambiar committed:
Merge branch 'master' of https://github.com/input-output-hk/dapps-certification-web into PLT-6763

amnambiar committed:
PLT-6763 fixed merge conflict

amnambiar committed:
PLT-6763 Resolved incorrect code after commit resolve

bogdan-manole reviewed (changes requested):
The report upload schema is not in accordance with the latest BE `AutditorInput` expected schema. @RSoulatIOHK @amnambiar for any changes in BE please comment in order to make the changes. 
I will write down what BE expects for the moment and what is different in FE :
- ~~`certificationLevel` is expected to be a `number` now, not a `string`. I'm looking in the `reportUpload.schema.tsx`, is the transformation to the number made later?~~
- `summary` is not required on the BE
- same for `disclaimer` 
- `subject` matches `^[A-Za-z0-9_]{1,64}$` ( I know it fits the current one, but let's keep the same regex on both sides.
- `discord` is different on BE (`^.{3,32}#[0-9]{4}$`) but I guess that's where the modification should be changed, not in the FE
- the `reportUrl` is different and has a pdf or txt mandatory suffix, though,  a valid pdf URL could not necessarily include the `pdf/txt` extension. Should we enforce this on BE as well?
- `repository`  in CIP has no validation pattern, in FE has `/^(?:https?:\/\/)?(?:www\.)?github\.com\/[\w-]+\/[\w.-]+$/`.  @RSoulatIOHK what solution should we choose:
   - we can keep it as it is in both places
   - we can remove the validation in FE
   - we can add the validation in the BE

UPDATE 1: I verified `certificationLevel` it's transmitted as number, it's ok

UPDATE 2: 
 - `social.website`, `social.github`, `social.link` gets the same value from the `website` form field
 - the `social.link` does not exist in CIP or BE
 - `social.github` on the backend is just the account name with pattern `^(?=.{1,39}$)[a-zA-Z0-9]+(-[a-zA-Z0-9]+)*$` 
 UPDATE 3:
 - `social.discord` is not sent into the request

bogdan-manole reviewed (changes requested):
Optional fields are sent as empty strings instead of null/key-missing. (Eg. `twitter`, `github`, logo etc)

<img width="971" alt="image" src="https://github.com/input-output-hk/dapps-certification-web/assets/3176255/b2eb4696-83bf-4faa-96e8-7925430170d7">
<img width="281" alt="image" src="https://github.com/input-output-hk/dapps-certification-web/assets/3176255/6d2dd355-716d-4a5b-8ad7-b5b32ce36348">

RSoulatIOHK commented:
> summary is not required on the BE

It should be, sorry for missing that

> same for disclaimer

Same, it should be.

> subject matches ^[A-Za-z0-9_]{1,64}$ ( I know it fits the current one, but let's keep the same regex on both sides.

Agreed

> discord is different on BE (^.{3,32}#[0-9]{4}$) but I guess that's where the modification should be changed, not in the FE

I don't have a strong opinion but let's take the one from CIP-0096.

> the reportUrl is different and has a pdf or txt mandatory suffix, though, a valid pdf URL could not necessarily include the pdf/txt extension. Should we enforce this on BE as well?

Good point, no then. Let's have the most open solution in place and remove the pdf/txt mandatory suffixes and maybe the report could be a json file as well so let's not complicate at the moment.

> repository in CIP has no validation pattern, in FE has /^(?:https?:\/\/)?(?:www\.)?github\.com\/[\w-]+\/[\w.-]+$/. 

Yes but we have one that is imposed by our tool only being able to verify github repos at the moment.

RSoulatIOHK commented:
> * `social.website`, `social.github`, `social.link` gets the same value from the `website` form field

To fix then.

> * the `social.link` does not exist in CIP or BE

Good catch, it got removed.

amnambiar committed:
PLT-6763 Schema corrections and payload fixes

amnambiar commented:
> subject matches ^[A-Za-z0-9_]{1,64}$ ( I know it fits the current one, but let's keep the same regex on both sides.

Updated the FE validation to be `.required(...).max(64, ...).matches(/^[A-Za-z0-9_]+$/, ....` just to ensure the form errors are thrown properly


_Other fixes done_ - 
- remove mandatory json/pdf extension checks at the reportUrl
- update social.github pattern to ^(?=.{1,39}$)[a-zA-Z0-9]+(-[a-zA-Z0-9]+)*$ rather than url
- social has properties - twitter, github, contact*, website*, discord ; fixed value mapping in these fields
- ensure optional fields not sent as empty-strings in payload; they are either null or not existing
- Github - field was missing in Auditor Report Upload, added it there.

Cc: @bogdan-manole @RSoulatIOHK

amnambiar committed:
Merge branch 'master' of https://github.com/input-output-hk/dapps-certification-web into PLT-6763

amnambiar committed:
PLT-6763 Removed Subject from Certification Metadata as part of updates to auditor profile in revamp

amnambiar commented:
Created https://input-output.atlassian.net/browse/PLT-7608 to ensure/handle fixes in revamp

amnambiar committed:
PLT-6763 Temporary hack in place to get this working with BE PR#90#94

bogdan-manole reviewed (approved):
lgtm

RSoulatIOHK reviewed (approved):
Looks good
